### PR TITLE
QA-394 - WAND

### PR DESCRIPTION
**Pull request by alexbakharew**
### Scope & Purpose

Extend existing tests for WAND feature

- [ ] :hankey: Bugfix
- [X] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [X] Enterprise PR: https://github.com/arangodb/enterprise/pull/1257
- [X] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/QA-394
- [ ] Design document:

### CODE REVIEW

File: tests/js/common/test-data/search/docs/generate_ii_sa_dataset.py [hunk 1]
```diff
@@ -102,52 +115,36 @@ def main():
...
+    #     _from, _to =  random.choices(ids, k=2)
+    #     edges.append(
+    #         {
+    #             "_key": str(i),
+    #             "_from": _from,
+    #             "_to": _to,
+    #             "Personal": generatePersonalData()
+    #         }
+    #     )
+    # resultCollection[collectionName] = edges
```
Comment (MBkkt) [inline tests/js/common/test-data/search/docs/generate_ii_sa_dataset.py:147]: Is it correct?

File: tests/js/common/test-data/search/docs/generate_ii_sa_dataset.py [hunk 2]
```diff
@@ -36,20 +36,33 @@ def generatePlaces():
...
     res["Geburtsdatum"] = generatePersonalData.fake.date()
-    res["Setzt"] = []
+    res["Setzt"] = ["1", "2", "3"]
 
-    numOfSetzt = random.randint(0, 8)
-    for _ in range(numOfSetzt):
-        res["Setzt"].append(generatePlaces())
+    # numOfSetzt = random.randint(0, 8)
+    # for _ in range(numOfSetzt):
+    #     res["Setzt"].append(generatePlaces())
```
Comment (MBkkt) [inline tests/js/common/test-data/search/docs/generate_ii_sa_dataset.py:60]: Same question about these comments

**Review by MBkkt — APPROVED** [review]
Except comment out generate_ii_sa_dataset. It looks like draft

File: tests/js/common/test-data/search/docs/generate_ii_sa_dataset.py (hunk 1 above)
Comment (alexbakharew) [reply tests/js/common/test-data/search/docs/generate_ii_sa_dataset.py:147]: Thanks for noticing. This should be reverted

File: tests/js/common/aql/aql-search-optimize-top-k.js [hunk 3]
```diff
@@ -44,18 +50,18 @@ function testOptimizeTopK() {
...
 
       db._createView("v_search_tfidf", "arangosearch", {
-        links: {"c": {fields: {"f": {}}}},
+        links: { "c": { fields: { "f": {}, "a": {}, "x": {} } } },
         optimizeTopK: ["TFIDF(@doc) DESC"]
       });
 
-      db.c.ensureIndex({name: "i_bm25", type: "inverted", fields: ["f"], optimizeTopK: ["BM25(@doc) DESC"]});
-      db._createView("v_alias_bm25", "search-alias", {indexes: [{collection: "c", index: "i_bm25"}]});
+      db.c.ensureIndex({ name: "i_bm25", type: "inverted", fields: ["f", "a", "x"], optimizeTopK: ["BM25(@doc) DESC"] });
```
Comment (Dronplane) [inline tests/js/common/aql/aql-search-optimize-top-k.js:57]: Data has fields  f, **c**, x but index indexes f, **a**, x - is that intentional?

File: tests/js/common/aql/aql-search-optimize-top-k.js [hunk 4]
```diff
@@ -23,18 +23,24 @@
...
 const jsunity = require("jsunity");
+const internal = require("internal");
 const arangodb = require("@arangodb");
 const db = arangodb.db;
-const isEnterprise = require("internal").isEnterprise();
+const isEnterprise = internal.isEnterprise();
+const dbName = "testOptimizeTopK";
 
-function testOptimizeTopK() {
+function testOptimizeTopKEnterprise() {
```
Comment (Dronplane) [inline tests/js/common/aql/aql-search-optimize-top-k.js:32]: Is it possible to move enterprise part to the enterprise repo?

File: tests/js/common/aql/aql-search-optimize-top-k.js [hunk 5]
```diff
@@ -65,42 +71,213 @@ function testOptimizeTopK() {
...
       for (const v of names) {
-        const query = "FOR d IN " + v + " SEARCH d.f == 'a' OPTIONS {waitForSync:true} LIMIT 10 RETURN d";
-        print(query);
+        const query = "FOR d IN " + v + " SEARCH d.f == 'a' OPTIONS {waitForSync:true} LIMIT 100 RETURN d";
         let res = db._query(query).toArray();
-        assertEqual(res.length, 10);
+        assertEqual(res.length, 100);
+      }
+
+      for (const v of names) {
```
Comment (Dronplane) [inline tests/js/common/aql/aql-search-optimize-top-k.js:79]: What are we testing here? One round without WAND to check sanity is ok. But what is behind repeating queries with different LIMIT?

File: tests/js/common/aql/aql-search-optimize-top-k.js [hunk 6]
```diff
@@ -65,42 +71,213 @@ function testOptimizeTopK() {
...
+    tearDownAll: function () {
+      db._useDatabase('_system');
+      try { db._dropDatabase(dbName); } catch (err) { }
     },
 
     testWithTFIDF: function () {
       for (const v of names) {
-        const query = "FOR d IN " + v + " SEARCH d.f == 'a' SORT TFIDF(d) LIMIT 10 RETURN d";
-        print(query);
+        const query = "FOR d IN " + v + " SEARCH d.f == 'a' SORT TFIDF(d) LIMIT 100 RETURN d";
```
Comment (Dronplane) [inline tests/js/common/aql/aql-search-optimize-top-k.js:99]: WAND would not be used here as default sorting is ASC. I guess DESC was expected here?

File: tests/js/common/aql/aql-search-optimize-top-k.js [hunk 7]
```diff
@@ -65,42 +71,213 @@ function testOptimizeTopK() {
...
+        let res = db._query(query).toArray();
+        assertEqual(res.length, 1000);
       }
     },
 
     testWithBM25: function () {
       for (const v of names) {
-        const query = "FOR d IN " + v + " SEARCH d.f == 'a' SORT BM25(d) LIMIT 10 RETURN d";
-        print(query);
+        const query = "FOR d IN " + v + " SEARCH d.f == 'a' SORT BM25(d) LIMIT 100 RETURN d";
```
Comment (Dronplane) [inline tests/js/common/aql/aql-search-optimize-top-k.js:119]: SORT DESC as above?

File: tests/js/common/aql/aql-search-optimize-top-k.js (hunk 3 above)
Comment (alexbakharew) [reply tests/js/common/aql/aql-search-optimize-top-k.js:57]: Yes

File: tests/js/common/aql/aql-search-optimize-top-k.js (hunk 5 above)
Comment (alexbakharew) [reply tests/js/common/aql/aql-search-optimize-top-k.js:79]: You have asked about something like this during previous review:
```
Here we need to test potential issues with AQL batch switching. And it is needed in PR runs. Benchmarks are not the question here. Is it possible to just auto-generate 2000-3000 documents and run query that will return around 1500? If do batch-inserts it should be quite fast.
```

Comment (alexbakharew) [reply tests/js/common/aql/aql-search-optimize-top-k.js:79]: Ah, forgot to add scorers. Will fix

File: tests/js/common/aql/aql-search-optimize-top-k.js (hunk 3 above)
Comment (alexbakharew) [reply tests/js/common/aql/aql-search-optimize-top-k.js:57]: But I will change it

**Review by Dronplane — APPROVED** [review]
LGTM

**Review by KVS85 — APPROVED** [review]

**Pull request merged**